content/blog/2024-12-03-tmil-11/index.md

```diff
@@ -0,0 +1,227 @@
+
+## Color
+
+When we published the update for [October, 2024](/blog/tmix-10/), we had already been discussing starting a new color-focused crate, but hadn't yet created a repository.
+
+We may have a dedicated blog post about the `color` crate in the future and laying out some of the motivations, but we'll go over it a bit here.
+
+Linebender, among other things, believes in helping build solid foundations for people building user interfaces and working in graphics, in the Rust programming language (and on GPUs).
+Upon seeing a problem, we prefer to produce a focused crate that solves that problem and that can be used by others within the Rust ecosystem.
+Color is one such area.
+It is much harder than it may appear at first, with many complicating factors like what should be done by a basic color struct, what color spaces to support, and whether or not to support color formats used in print (like CMYK).
+
+We looked around at a wide variety of crates in various stages of maintenance and development, their feature sets, the "weight" of those crates, and how closely they fit in with what we would need in Vello, Tiny Skia, Resvg, SVG Types, Masonry, and Xilem.
+We decided to do a crate based on the [CSS Color Module Level 4] draft specification, with support for a limited number of additional color spaces.
+
+We released an alpha of this new crate as [Color 0.1.0][] on November 21, 2024.
+The documentation lays out more of the scope, goals, and main types provided by the crate.
+
+<style>
+div.gradient-demo { height: 50px }
+#browser { background: linear-gradient(to right in oklab, lime, blue) }
+#ours { background: linear-gradient(to right, color(srgb 0 1 0) 0%, color(srgb -0.22411254 0.92055106 0.4044056) 12.5%, color(srgb -0.28811926 0.8390731 0.5539247) 25%, color(srgb -0.31709152 0.66550463 0.7479822) 50%, color(srgb -0.26123333 0.45950305 0.88687235) 75%, color(srgb -0.19183734 0.3212008 0.94574267) 87.5%, color(srgb 0 0 1) 100%) }
+</style>
+
+<figure>
+
+<div class='gradient-demo' id='ours'></div>
+
+<figcaption>
+
+Color can convert gradients between color spaces.
+This is a gradient from "lime" to "blue" in the Oklab color space, converted to sRGB by Color.
+
+</figcaption>
+</figure>
+
+<figure>
+
+<div class='gradient-demo' id='browser'></div>
+
+<figcaption>
+
+This is a gradient from "lime" to "blue" in Oklab, as generated by your browser.
+This is the reference for the above gradient, and should appear identical.
+
+</figcaption>
+</figure>
+
+Since then, we have begun the work of making Peniko and Vello use the Color crate and improving it to support that usage.
+Early next year, we hope to use it within SVG Types, Resvg, and Tiny Skia.
+
+We will be publishing a 0.2.0 release within the next month and encourage people to give it a try.
+The 0.2.0 release will include some additional (non-CSS 4) color spaces like ACEScg, OkHsl, and OkHsv as well as things that we found useful when using it in Peniko and Vello.
```

Comment:
OkHsl and OkHsv didn't make the release.

```suggestion
The 0.2.0 release will include the (non-CSS 4) color space ACEScg, as well as things that we found useful when using it in Peniko and Vello.
```

Reply:
This is part of why I didn't like us being specific about "what we'll do in December", when it turns out not to be what we'll do.